### PR TITLE
Stale Checking Logic Fixes

### PR DESCRIPTION
**Pull request by h3h**
* Fix crash in stale checking logic
* Remove bad method call
* Ensure unicode strings from local txt files
* Mark up channel names containing hyphens

### CODE REVIEW

File: destalinator.py [hunk 1]
```diff
@@ -120,18 +120,21 @@ def stale(self, channel_name, days):
             # self.debug("Not checking if {} is stale -- it's too new".format(channel_name))
             return False
         messages = self.get_messages(channel_name, days)
-        messages = [
+        # return True (stale) if none of the messages match the criteria below
```
Comment (zeckalpha) [inline destalinator.py]: _gasp_ comments!

File: destalinator.py [hunk 2]
```diff
@@ -120,18 +120,21 @@ def stale(self, channel_name, days):
...
+        return not any(
```
Comment (zeckalpha) [inline destalinator.py]: If we use the not any, moving the if clause to the item clause makes more sense type wise, e.g. 

```python
return not any(
            x.get("user") not in self.config.ignore_users
            and (
               # the message must have text that doesn't include ignored words
                (x.get("text") and b":dolphin:" not in x.get("text").encode('utf-8', 'ignore'))
                # or the message must have attachments
                or x.get("attachments")
            )
       for x in messages
)
```

but I think what you have works.

File: destalinator.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def add_slack_channel_markup_item(self, item):
         return self.slacker.add_channel_markup(item.group(1))
 
     def add_slack_channel_markup(self, text):
-        marked_up = re.sub(r"\#(\w+)", self.add_slack_channel_markup_item, text)
+        marked_up = re.sub(r"\#([\w-]+)", self.add_slack_channel_markup_item, text)
```
Comment (zeckalpha) [inline destalinator.py]: I think underscores are allowed as well.

File: destalinator.py (hunk 2 above)
Comment (h3h) [inline destalinator.py]: Great point. A much more pythonic representation after this refactor.

File: destalinator.py (hunk 3 above)
Comment (h3h) [inline destalinator.py]: `\w` includes underscores, happily. But not hyphens, sadly.

Comment (h3h) [inline destalinator.py]: Though this reveals some underlying weaknesses of using this character class:
* Uppercase letters are _not_ supported in channel names, but _are_ included in `\w`
* `\w` isn't the best explanation here; we should likely use a full character class to describe the allowable set, e.g. `[a-z_-]`, once I hear back from the Slack team on what “non-Latin characters” might mean. 😄

**Comment by h3h** [comment]
- [ ] @h3h Refactor generator for `any()` to avoid `if` filter
- [ ] @h3h Explode channel-detecting regex to include full literal character class

**Pull request merged**